docs/blog/2019-01-04-using-react-context-api-with-gatsby/index.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Using React Context API with Gatsby
+author: Muhammad Muhsin
+date: 2019-01-04
+tags:
+  - "React"
+---
+
+You often feel the unsettling flash of a bright phone screen while relaxing in a dimly lit room. The "dark mode" ends it, by switching background and foreground colors to reduce eye strain. I decided to add this to my boutique web agency [Laccadive IO's](https://laccadive.io/) new Gatsby-based site.
+
+>
+> One of the few types of alternative theme that adds real value to users is a low light intensity “night mode” theme. Not only is it easier on the eyes when reading in the dark, but it also reduces the likelihood of migraine and the irritation of other light sensitivity disorders. As a migraine sufferer, I’m interested!
+>
+> [Heydon Pickering](https://inclusive-components.design/a-theme-switcher/)
+
+While discussing the different ways to implement this, I came up with React’s new Context API as a possible implementation strategy. Having worked with Context API before, I set to code. However, I soon realized you will not see a ‘root’ in a Gatsby app like an app built with Create React App.
+
+After Googling a bit, I came across the Gatsby Browser APIs. Specifically, the [wrapRootElement](https://www.gatsbyjs.org/docs/browser-apis/#wrapRootElement) API was what I was looking for. This essentially allows you to wrap your root element with a Provider from Redux or Context for instance. Using this, I managed to achieve dark mode for Laccadive IO.
+
+The following is a detailed instruction on how you too can achieve a similar result with Gatsby and React Context!
```

Comment:
Let's do a deep dive into how this feature was actually implemented step by step using React Context, Gatsby, and a Theme Provider to implement a dark mode UI!